Zipped plugins (11207) (#117)
endless-sky/endless-sky-wiki: wiki/CreatingPlugins.md

	  ...
```

> Alternatively, you can use your plugin as a zip file. The plugin's internal structure is the same as before:

Author: ```suggestion
Starting with **0.10.13**, you can also use your plugin as a zip file. The plugin's internal structure is the same as before:
```

> Starting with **0.10.13**, you can also use your plugin as a zip file. The plugin's internal structure is the same as before:

Reviewer: Isn't the convention in this sort of file to use
```suggestion
Beginning in **0.10.13**, you can also use your plugin as a zip file. The plugin's internal structure is the same as before:
```

```
plugins/
|-- example-plugin.zip/
|   |-- data/
|   |   |-- jobs.txt
|   |	 ...
|   |-- icon.png
|   |-- plugin.txt
|   |-- copyright
|	 ...
|-- other-plugin/
	  ...
```

> The game also supports plugins that are zipped together with their directory, though this only works if the zip file's name is same as the plugin directory's name (in this case, `example-plugin.zip` and `example-plugin`):

Author: ```suggestion
The game also supports plugins that are zipped together with their directory:
```

```
plugins/
|-- example-plugin.zip/

> |   |-- example-plugin/

Reviewer 2: I feel that it makes sense to only support the version without this intermediate directory. Just in case someone just wants to rename the zip-file, and because the extra directory layer doesn't add much.

Supporting only the "without intermediate directory" version makes the code a little bit simpler, and makes renames of zipfiles also simpler. (And people creating zip-files should just add the relevant compression options to make this work.)

Author: The issue is that nearly all github artifacts have the intermediate directory by default, including high-dpi and pretty much everything in the plugin repository. I could look into supporting any plugin with a single top-level directory, though.

Author: I changed it so any top-level directory is supported, so renaming the zip file wouldn't be an issue anymore.

Reviewer 2: That also works for me.
